app/src/main/java/com/example/ar_reshare/BackendController.java

```diff
@@ -273,6 +288,253 @@ public void onFailure(Call<ResponseBody> call, Throwable t) {
         return false;
     }
 
+
+    public static boolean createConversation(Integer listingID, BackendCallback callback){
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+
+        String bodyString =
+                String.format("{\n  \"listingID\": \"%d\"\n}",
```

Comment:
Hi Lingtao, I was wondering if you could please replace this, with the method @HellinSun found. That is, using a _JsonObject_. It is a bit more readable and neater then! 😄

Reply:
I have now replaced all body string creating with JSON object related methods, Thanks Artur!